Error while running dependabot

There seems to be an error when starting the configuration, but I can't find what...

### Pipeline YAML

```trigger: none # Disable CI trigger

pool:
  vmImage: 'ubuntu-latest' # requires macos or ubuntu (windows is not supported)

steps:
- task: dependabot@1
  inputs:
    packageManager: 'pip'
    directory: '/arena'
    failOnException: false
    setAutoComplete: true
    targetRepositoryName: 'api-arena'
- task: dependabot@1
  inputs:
    packageManager: 'npm'
    directory: '/'
    failOnException: false
    setAutoComplete: true
    targetRepositoryName: 'ui-forge'
```

### Result log

```[...]
5bc9af957871: Pull complete
46ea4c3c98d2: Pull complete
Digest: sha256:a182c561bb28cac4d30edfaee21ebed3f06ec83f25ad1efec9324c67e7cb8d95
Status: Downloaded newer image for tingle/dependabot-azure-devops:0.3
warning: parser/current is loading parser/ruby26, which recognizes
warning: 2.6.7-compliant syntax, but you are running 2.6.6.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
/home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:444:in `rescue in default_branch_for_repo': Dependabot::RepoNotFound (Dependabot::RepoNotFound)
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:440:in `default_branch_for_repo'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:70:in `commit'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:359:in `_full_specification_for'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:174:in `_fetch_repo_contents'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:159:in `repo_contents'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:102:in `fetch_file_if_present'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-python-0.142.1/lib/dependabot/python/file_fetcher.rb:116:in `pipfile'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-python-0.142.1/lib/dependabot/python/file_fetcher.rb:64:in `pipenv_files'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-python-0.142.1/lib/dependabot/python/file_fetcher.rb:41:in `fetch_files'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:64:in `files'
	from ./update-script.rb:172:in `<main>'
/home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/clients/azure.rb:231:in `get': Dependabot::Clients::Azure::NotFound (Dependabot::Clients::Azure::NotFound)
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/clients/azure.rb:57:in `fetch_default_branch'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:442:in `default_branch_for_repo'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:70:in `commit'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:359:in `_full_specification_for'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:174:in `_fetch_repo_contents'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:159:in `repo_contents'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:102:in `fetch_file_if_present'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-python-0.142.1/lib/dependabot/python/file_fetcher.rb:116:in `pipfile'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-python-0.142.1/lib/dependabot/python/file_fetcher.rb:64:in `pipenv_files'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-python-0.142.1/lib/dependabot/python/file_fetcher.rb:41:in `fetch_files'
	from /home/dependabot/dependabot-script/vendor/ruby/2.6.0/gems/dependabot-common-0.142.1/lib/dependabot/file_fetchers/base.rb:64:in `files'
	from ./update-script.rb:172:in `<main>'
Using 'bvacx.visualstudio.com' as hostname
Fetching pip dependency files for bvacx/PCX/_git/api-arena
Targeting 'default' branch under '/arena' directory
Using 'auto' versioning strategy
##[error]The process '/usr/bin/docker' failed with exit code 1```

--- a/src/extension/task/models/IDependabotUpdate.ts
+++ b/src/extension/task/models/IDependabotUpdate.ts
@@ -16,11 +16,11 @@ export interface IDependabotUpdate {
   /**
    * Customize which updates are allowed.
    */
-  allow?: IDependabotAllowDependency[] | string;
+  allow?: string;
   /**
    * Ignore certain dependencies or versions.
    */
-  ignore?: IDependabotIgnoreDependency[] | string;
+  ignore?: string;
   /**
    * 	Limit number of open pull requests for version updates.
    */
@@ -41,16 +41,6 @@ export interface IDependabotUpdate {
    * The milestone to associate pull requests with.
    */
    milestone?: string;
-}
-
-export interface IDependabotAllowDependency {
-  dependencyName: string;
-  dependencyType: string;
-}
-
-export interface IDependabotIgnoreDependency {
-  dependencyName: string;
-  versions: string[];
 }
 
 export interface IDependabotUpdateSchedule {

--- a/src/extension/task/utils/parseConfigFile.ts
+++ b/src/extension/task/utils/parseConfigFile.ts
@@ -69,8 +69,8 @@ export default function parseConfigFile(): IDependabotUpdate[] {
       milestone: update["milestone"],
 
       // Convert to JSON and shorten the names as required by the script
-      allow: updates["allow"] ? JSON.stringify(updates["allow"]) : undefined,
-      ignore: updates["ignore"] ? JSON.stringify(updates["ignore"]) : undefined,
+      allow: update["allow"] ? JSON.stringify(update["allow"]) : undefined,
+      ignore: update["ignore"] ? JSON.stringify(update["ignore"]) : undefined,
     };
 
     if (!dependabotUpdate.packageEcosystem) {

--- a/src/extension/task/utils/parseConfigFile.ts
+++ b/src/extension/task/utils/parseConfigFile.ts
@@ -28,6 +28,23 @@ export default function parseConfigFile(): IDependabotUpdate[] {
     throw new Error("Invalid dependabot config object");
   }
 
+  const rawVersion = config["version"];
+  let version = -1;
+
+  // ensure the version has been specified
+  if(!!!rawVersion) throw new Error("The version must be specified in dependabot.yml");
+
+  //try convert the version to integer
+  try{
+    version = parseInt(rawVersion, 10);
+  }
+  catch(e) {
+    throw new Error("Dependabot version specified must be a valid integer");
+  }
+
+  //ensure the version is == 2
+  if(version !== 2) throw new Error("Only version 2 of dependabot is supported. Version specified: " + version);
+
   var updates: IDependabotUpdate[] = [];
 
   //check the updates parsed

